ui/src/components/References/GroupReference.tsx

```diff
@@ -53,20 +56,24 @@ export default function GroupReference({ flag }: GroupReferenceProps) {
         </div>
       </button>
       <div className="absolute right-5 flex flex-row">
-        {gang.invite ? (
+        {gang.invite && status !== 'loading' ? (
           <button
             className="small-button bg-red text-white dark:text-black"
             onClick={reject}
           >
             Reject
           </button>
         ) : null}
-        <button
-          className="small-button ml-2 bg-blue text-white dark:text-black"
-          onClick={group ? open : join}
-        >
-          {button.text}
-        </button>
+        {status === 'loading' ? (
+          <LoadingSpinner />
+        ) : (
```

Comment:
May also want to handle the poke erroring immediately case and display an errored but disabled button.